### PR TITLE
docs: updating readme with guides for how to generate tests

### PR DESCRIPTION
**Pull request by lenareed**
Guides for how to iterate on prompt and how to write assertions

Also included explicit instructions on how to open the extension.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,12 +28,47 @@ automatically navigate the app and reach your desired outcome.
...
    ![installation-instruction](installation-instruction.png)
 5. Upload your mobile app file via [Sauce Labs App Management](https://app.saucelabs.com/app-management).
-6. Describe your test goal in plain English.
-7. Click "Generate Test" to let ScriptIQ navigate your app on a Sauce Labs real device.
-8. Review the test results and iterate on your test goals as needed.
+6. Click ScriptIQ icon in Activity bar.
+
+   ![activity-bar-icon](activity-bar-icon.png)
+
+7. Input and save credentials (only need to do once unless new credentials necessary).
```
Comment (alexplischke) [inline README.md:35]: (style) Just "flows" a bit better for me personally:
```suggestion
7. Input and save credentials (do this once, unless new credentials are necessary).
```

File: README.md [hunk 2]
```diff
@@ -28,12 +28,47 @@ automatically navigate the app and reach your desired outcome.
...
 
 ScriptIQ brings the power of AI to mobile app testing, making it easier, and more accessible than ever.
 
 For more information and documentation, please visit the ScriptIQ documentation (coming soon).
+
+## Prompt Generation Hints
+
+ScriptIQ is powered by Large Language Models (LLMs). LLMs are very powerful, but sometimes it takes a couple of tries to create a prompt that the model can understand and interpret correctly. Therefore it might take multiple attempts to write a goal which generates the expected test. If your first attempt does not result in the desired outcome, and you are unsure why this is, you can refer to these potential issues with your prompt:
+
+- **Unclear word choice**: Consider whether or not the goal you created could be interpreted in multiple ways. If so, consider rewriting or adding clarifying statements.
```
Comment (alexplischke) [inline README.md:50]: (style) Just for brevity:
```suggestion
- **Unclear word choice**: Consider whether the goal you created could be interpreted in multiple ways. If so, consider rewriting or adding clarifying statements.
```

File: README.md [hunk 3]
```diff
@@ -28,12 +28,47 @@ automatically navigate the app and reach your desired outcome.
...
+3. Update the goal with this additional context and run again: ![ScriptIQ example goal which did work](media/imgs/example_scriptiq_correct.png)
+4. The updated goal is now accurately finding the login button and clicking it.
+
+Those are just some examples of why a goal might not work and how to fix it. Feel free to be creative!
+
+## How Do Assertions work?
+
+Once you've started generating tests, you might want to more quickly determine if your test succeeded or failed, this is where the _Assert Inputs_ fields come in. Assert Inputs are descriptions of screens which are used to evaluate the test generated. When assertions are provided, the LLM will determine if the assertion matches what is on the screen for each assertion provided against every screen visited. These assertions can describe the entire screen (e.g. login screen) or describe an element on the screen (e.g. page with checkout button). These assertions are _not_ used to generate test steps, any intent needed to drive the test generation must be included in the _Test Goal_ field. Below you can see examples of assertions and an example output.
```
Comment (alexplischke) [inline README.md:66]: This is a single line with 758 characters 😅 Would it be possible to break it down into 80 character chunks? Last time I checked with the team, that seemed like that preferred line length, especially when you try to place windows side by side. Easier to view them inside editors that don't have word-wrap enabled. Also easier to review any changes made in the future!

PS: This is true for some of the other lines above as well 🙏

File: README.md [hunk 4]
```diff
@@ -28,12 +28,47 @@ automatically navigate the app and reach your desired outcome.
...
+
+- **Unclear word choice**: Consider whether or not the goal you created could be interpreted in multiple ways. If so, consider rewriting or adding clarifying statements.
+- **Incorrect path taken**: ScriptIQ uses common sense reasoning to generate the test using the intent of your goal. But sometimes ScriptIQ takes the wrong path, making an assumption about the best way to accomplish your goal which is incorrect. When an incorrect step is taken, consider defining the correct action explicitly in your goal. For example, if ScriptIQ attempts to login when you do not require it to login, you can add "do not login" to your goal.
+- **Context provided in assertions**: The assertion statements do not guide ScriptIQ towards accomplishing the goal, they only check whether or not the descriptions provided in the assertion match seen screens. If you've only included necessary context in the assertion fields you'll need to add it to the goal.
```
Comment (alexplischke) [inline README.md:52]: Same case for brevity:
```suggestion
- **Context provided in assertions**: The assertion statements do not guide ScriptIQ towards accomplishing the goal, they only check whether the descriptions provided in the assertion match seen screens. If you've only included necessary context in the assertion fields you'll need to add it to the goal.
```

File: README.md (hunk 3 above)
Comment (lenareed) [reply README.md:66]: Yes! I can do that!

**Review by alexplischke — APPROVED** [review]
